src/ifcopenshell-python/ifcopenshell/validate.py

```diff
@@ -568,9 +568,9 @@ def to_string_header_entity(header_entity):
     """Recreate IFC header string representation, like FILE_NAME(...)"""
     
     # Prefer native .toString() if available (native IfcOpenShell wrapper)
-    if hasattr(header_entity, 'toString'):
+    if isinstance(header_entity, W.HeaderEntity):
         return header_entity.toString()
-    elif hasattr(header_entity, '_fields'):
+    elif isinstance(header_entity, tuple):
```

Comment:
```suggestion
    elif hasattr(header_entity, '_fields'):
    # namedtuple
```

Ok,. now I remember, isinstance doesn't work well with namedtuple, because it does some sort of wonky code generation. Sorry, then let's keep that line as is, because just tuple is even more misleading.

Reply:
Oops, I'm just reading this. See comment above 
https://github.com/IfcOpenShell/IfcOpenShell/pull/6650#discussion_r2073853817

> simple_spf.file doesn't have a distinctive type for header entity (tuple comes the closest); this

What about 
```
if isinstance(header_entity, W.HeaderEntity)
   ... 
elif hasattr(header_entity, '_fields'):
   .. 
else
 ..
```
?

Reply:
I agree good compromise

Reply:
[Implemented](https://github.com/IfcOpenShell/IfcOpenShell/pull/6650/commits/50e348507afcbcce4895745ab987863b617f832f)